EpiAware/test/test_inference-methods.jl

```diff
@@ -0,0 +1,135 @@
+@testitem "Testing _run_manypathfinder function" begin
+    using Turing, Pathfinder
+    @model function test_model()
+        x ~ Normal(0, 1)
+        y ~ Normal(x, 1)
+    end
+
+    mdl = test_model()
+
+    # Test case 1
+    @testset "Test case 1" begin
+        nruns = 10
+        ndraws = 100
+        maxiters = 50
+
+        pfs = EpiAware._run_manypathfinder(
+            mdl; nruns = nruns, ndraws = ndraws, maxiters = maxiters)
+
+        @test length(pfs) == nruns
+        @test all(p -> p isa Union{PathfinderResult, Symbol}, pfs)
+    end
+
+    # Test case 2
+    @testset "Test case 2" begin
+        nruns = 5
+        ndraws = 50
+        maxiters = 100
+
+        pfs = EpiAware._run_manypathfinder(
+            mdl; nruns = nruns, ndraws = ndraws, maxiters = maxiters)
+
+        @test length(pfs) == nruns
+        @test all(p -> p isa Union{PathfinderResult, Symbol}, pfs)
+    end
+end
+@testitem "Testing _continue_manypathfinder! function" begin
+    using Turing, Pathfinder
+
+    @testset "Test case 1" begin
+        @model function test_model()
+            x ~ Normal(0, 1)
+            y ~ Normal(x, 1)
+        end
+
+        mdl = test_model()
+
+        pfs = Vector{Union{PathfinderResult, Symbol}}([:fail, :fail, :fail])
+        max_tries = 3
+        nruns = 10
+        ndraws = 100
+        maxiters = 50
+
+        pfs = EpiAware._continue_manypathfinder!(
+            pfs, mdl; max_tries, nruns, ndraws, maxiters)
+
+        @test all(p -> p isa Union{PathfinderResult, Symbol}, pfs)
+    end
+
+    @testset "Test case 2" begin
+        @model function test_model()
+            x ~ Normal(0, 1)
+            y ~ Normal(x, 1)
+        end
+
+        mdl = test_model()
+
+        pfs = Vector{Union{PathfinderResult, Symbol}}([:fail, :fail, :fail])
+        max_tries = 3
+        nruns = 10
+        ndraws = 100
+        maxiters = 50
+
+        pfs = EpiAware._continue_manypathfinder!(
+            pfs, mdl; max_tries, nruns, ndraws, maxiters)
+
+        @test all(p -> p isa Union{PathfinderResult, Symbol}, pfs)
+    end
+end
+@testitem "Testing _get_best_elbo_pathfinder function" begin
+    using Pathfinder, Turing
+
+    @model function test_model()
+        x ~ Normal(0, 1)
+        y ~ Normal(x, 1)
+    end
+
+    mdl = test_model()
+    nruns = 10
+    ndraws = 100
+    maxiters = 50
+
+    pfs = EpiAware._run_manypathfinder(
+        mdl; nruns = nruns, ndraws = ndraws, maxiters = maxiters)
+
+    best_pf = EpiAware._get_best_elbo_pathfinder(pfs)
+    @test best_pf isa PathfinderResult
+end
+@testitem "Testing manypathfinder function" begin
+    using Turing, Pathfinder
+
+    @model function test_model()
```

Comment:
should this really live somewhere (like smoe kind of testing setup or something? It seems a bit mad to keep recreating it)

Reply:
Yeah! Thats a good point.

Reply:
I've made that an issue